code/modules/admin/verbs/modifyvariables.dm

```diff
@@ -297,212 +529,104 @@ var/list/forbidden_varedit_object_types = list(
-				return
-			O.vars[variable] = var_new
-			var_as_text = "matrix([matrix_text])"
-
-		if("reference")
-			var/var_new = input("Select reference:","Reference",O.vars[variable]) as null|mob|obj|turf|area in world
-			if(var_new==null) return
-			O.vars[variable] = var_new
-
-		if("mob reference")
-			var/var_new = input("Select reference:","Reference",O.vars[variable]) as null|mob in world
-			if(var_new==null) return
-			O.vars[variable] = var_new
-
-		if("file")
-			var/var_new = input("Pick file:","File",O.vars[variable]) as null|file
-			if(var_new==null) return
-			O.vars[variable] = var_new
-
-		if("icon")
-			var/var_new = input("Pick icon:","Icon",O.vars[variable]) as null|icon
-			if(var_new==null) return
-			O.vars[variable] = var_new
-
-		if("marked datum")
-			O.vars[variable] = holder.marked_datum
-
-	if(var_as_text == null)
-		var_as_text = "[O.vars[variable]]"
-	O.on_varedit(variable)
-	log_to_dd("### VarEdit by [src]: [O.type] [variable]=[html_encode("[var_as_text]")]")
-	log_admin("[key_name(src)] modified [original_name]'s [variable] to [var_as_text]")
-	message_admins("[key_name_admin(src)] modified [original_name]'s [variable] to [var_as_text]", 1)
-
-// Let's get this all in one place.
-// You'll need to take care of setting dir or iconizing the variable yourself once you've called this
-/proc/variable_to_type(var/variable)
-	var/class
-	if(isnull(variable))
-		to_chat(usr, "Unable to determine variable type.")
-		class = null
-	else if(isnum(variable))
-		to_chat(usr, "Variable appears to be <b>NUM</b>.")
-		class = "num"
-
-	else if(istext(variable))
-		to_chat(usr, "Variable appears to be <b>TEXT</b>.")
-		class = "text"
-
-	else if(isloc(variable))
-		to_chat(usr, "Variable appears to be <b>REFERENCE</b>.")
-		class = "reference"
-
-	else if(isicon(variable))
-		to_chat(usr, "Variable appears to be <b>ICON</b>.")
-		variable = "[bicon(variable)]"
-		class = "icon"
-
-	else if(istype(variable,/matrix))
-		to_chat(usr, "Variable appears to be <b>MATRIX</b>")
-		class = "matrix"
-
-	else if(istype(variable,/atom) || istype(variable,/datum))
-		to_chat(usr, "Variable appears to be <b>TYPE</b>.")
-		class = "type"
-
-	else if(istype(variable,/list))
-		to_chat(usr, "Variable appears to be <b>LIST</b>.")
-		class = "list"
-
-	else if(istype(variable,/client))
-		to_chat(usr, "Variable appears to be <b>CLIENT</b>.")
-		class = "cancel"
-
-	else if(ispath(variable))
-		to_chat(usr, "Variable appears to be <b>PATH</b>.")
-		class = "path"
-
-	else if(isfile(variable))
-		to_chat(usr, "Variable appears to be <b>FILE</b>.")
-		class = "file"
+		if(VV_RESTORE_DEFAULT)
+			var_new = initial(O.vars[variable])
 
-	else
-		to_chat(usr, "Variable type is <b>UNKNOWN</b>.")
-		class = null
+		if(VV_TEXT)
+			var/list/varsvars = vv_parse_text(O, var_new)
+			for(var/V in varsvars)
+				var_new = replacetext(var_new,"\[[V]]","[O.vars[V]]")
 
-	return class
+	if(O.vv_edit_var(variable, var_new) == FALSE)

```

Comment:
I notice that in some places you just use the conditional off of the return value directly, while still having the same "rejected by the object" message. Is this intentional?